tests/suites/test_suite_psa_crypto_driver_wrappers.function

```diff
@@ -0,0 +1,243 @@
+    0x4a, 0x57, 0xa0, 0xba, 0x01, 0x20, 0x42, 0x08,
+    0x70, 0x97, 0x49, 0x6e, 0xfc, 0x58, 0x3f, 0xed,
+    0x8b, 0x24, 0xa5, 0xb9, 0xbe, 0x9a, 0x51, 0xde,
+    0x06, 0x3f, 0x5a, 0x00, 0xa8, 0xb6, 0x98, 0xa1,
+    0x6f, 0xd7, 0xf2, 0x9b, 0x54, 0x85, 0xf3, 0x20
+};
+
+uint8_t test_fake_output[] = "INJECTED OUTPUT";
+
+typedef enum
+{
+    EXPECT_FAILURE,
+    EXPECT_CORRECT_OUTPUT,
+    EXPECT_FAKE_OUTPUT,
+} expected_output_t;
+
+/* END_HEADER */
+
+/* BEGIN_DEPENDENCIES
+ * depends_on:MBEDTLS_PSA_CRYPTO_C:MBEDTLS_PSA_CRYPTO_DRIVERS:MBEDTLS_PSA_CRYPTO_DRIVER_TEST
+ * END_DEPENDENCIES
+ */
+
+/* BEGIN_CASE depends_on:MBEDTLS_ECDSA_C:MBEDTLS_ECDSA_DETERMINISTIC:MBEDTLS_ECP_DP_SECP256R1_ENABLED:MBEDTLS_SHA256_C */
+void ecdsa_sign( int force_status_arg,
+                 int fake_output,
+                 int expected_status_arg )
+{
+    psa_status_t force_status = force_status_arg;
+    psa_status_t expected_status = expected_status_arg;
+    psa_key_handle_t handle = 0;
+    psa_key_attributes_t attributes = PSA_KEY_ATTRIBUTES_INIT;
+    psa_algorithm_t alg = PSA_ALG_DETERMINISTIC_ECDSA( PSA_ALG_SHA_256 );
+    uint8_t signature[64];
+    size_t signature_length = 0xdeadbeef;
+    const uint8_t *expected_output;
+    size_t expected_output_length;
+    psa_status_t actual_status;
+
+    PSA_ASSERT( psa_crypto_init( ) );
+    psa_set_key_type( &attributes,
+                      PSA_KEY_TYPE_ECC_KEY_PAIR( PSA_ECC_CURVE_SECP_R1 ) );
+    psa_set_key_usage_flags( &attributes, PSA_KEY_USAGE_SIGN_HASH );
+    psa_set_key_algorithm( &attributes, alg );
+    psa_import_key( &attributes,
+                    test_secp256r1_key_data, sizeof( test_secp256r1_key_data ),
+                    &handle );
+
+    test_transparent_signature_sign_hash_hit = 0;
+    test_transparent_signature_sign_hash_status = force_status;
+    if( fake_output )
+    {
+        expected_output = test_driver_forced_output = test_fake_output;
+        expected_output_length = test_driver_forced_output_length =
+            sizeof( test_fake_output );
+    }
+    else
+    {
+        expected_output = test_signature_hash_32_with_secp256r1;
+        expected_output_length = sizeof( test_signature_hash_32_with_secp256r1 );
+    }
+
+    actual_status = psa_sign_hash( handle, alg,
+                                   test_hash_32, sizeof( test_hash_32 ),
+                                   signature, sizeof( signature ),
+                                   &signature_length );
+    TEST_EQUAL( actual_status, expected_status );
+    if( expected_status == PSA_SUCCESS )
+    {
+        ASSERT_COMPARE( signature, signature_length,
+                        expected_output, expected_output_length );
+    }
+    TEST_EQUAL( test_transparent_signature_sign_hash_hit, 1 );
+
+exit:
+    psa_reset_key_attributes( &attributes );
+    psa_destroy_key( handle );
+    PSA_DONE( );
+    test_transparent_signature_sign_hash_status = PSA_ERROR_NOT_SUPPORTED;
```

Comment:
With a structure grouping those variables, just reset the structure using the TEST_DRIVER_SIGNATURE_INIT macro probably.